### PR TITLE
[Merge] Make Teku working with Geth EE

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

A lot of API compatibility issues fixed

With these changes Teku works with [Geth EE ](https://github.com/ethereum/go-ethereum/pull/23607), successfully produces and imports blocks with `ExecutionPayload` composed by Geth EE

### Run instructions

1. Build and start Geth EE as described [here](https://notes.ethereum.org/@9AeMAlpyQYaAAyuj47BzRw/rkwW3ceVY)
   **NOTE**: as per spec the `GENESIS_GAS_LIMIT` should be `30 000 000` so make sure (or change) the Geth genesis file to have value `"gasLimit": "0x1C9C380"`.  With that value the block hash would be `0x3b8fb240d288781d4aac94d3fd16809ee413bc99294a085798a589dae51ddd4a`
2. Generate consensus genesis state:
```bash
> teku genesis mock \
  --eth1-block-hash 0x3b8fb240d288781d4aac94d3fd16809ee413bc99294a085798a589dae51ddd4a \
  --output-file ./local-genesis.ssz \ 
  --network=minimal \
  --Xnetwork-altair-fork-epoch=0 \
  --Xnetwork-merge-fork-epoch=0 \
  --validator-count=256
```
3. Run a Teku node:
```bash
> teku \
  --eth1-endpoints=http://localhost:8545 \
  --ee-fee-recipient-address=0xfe3b557e8fb62b89f4916b721be55ceb828dbd73 \
  --Xinterop-enabled=true \
  --Xinterop-number-of-validators=256 \
  --Xinterop-owned-validator-start-index=0 \
  --Xinterop-owned-validator-count=256 \
  --network=minimal \
  --Xnetwork-altair-fork-epoch=0 \ 
  --Xnetwork-merge-fork-epoch=0 \ 
  --p2p-enabled=false \
  --initial-state ./local-genesis.ssz
```

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ GENESIS_FORK_VERSION: 0x00000000
 GENESIS_DELAY: 604800
 
 # Genesis - Merge
-GENESIS_GAS_LIMIT: 30000000
+GENESIS_GAS_LIMIT: 10000000
```
Comment (tbenr) [inline util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml]: why `10000000` if spec says `30000000`? is it a Geth specific?

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java [hunk 2]
```diff
@@ -44,23 +47,22 @@ public Web3JExecutionEngineClient(String eth1Endpoint, Optional<String> maybeEeE
...
+  public SafeFuture<UInt64> preparePayload(PreparePayloadRequest request) {
+    Request<?, UInt64Web3jResponse> web3jRequest =
         new Request<>(
             "engine_preparePayload",
             Collections.singletonList(request),
             eeWeb3jService,
-            PreparePayloadWeb3jResponse.class);
-    return doRequest(web3jRequest);
+            UInt64Web3jResponse.class);
+    return doRequest1(web3jRequest).thenApply(UInt64Web3jResponse::getUInt);
```
Comment (tbenr) [inline services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java]: (nit) `doRequest1` name is odd, maybe we could find something better when we will have the final version of the APIs if this is still required.

**Review by tbenr — APPROVED** [review]
LGTM with some comments

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java (hunk 2 above)
Comment (Nashatyrev) [reply services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java]: Oh, yes, thanks! That was a working method name :)

File: util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml (hunk 1 above)
Comment (Nashatyrev) [reply util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml]: Hm, yep, that was a value from Geth. Need to find out what's the right value

**Comment by Nashatyrev** [comment]
Updated to the final interop spec version. Mostly `engine_preparePayload` return type reverted back to object instead of value

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java (hunk 2 above)
Comment (Nashatyrev) [reply services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/Web3JExecutionEngineClient.java]: That method was finally removed

File: util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml (hunk 1 above)
Comment (Nashatyrev) [reply util/src/main/resources/tech/pegasys/teku/util/config/configs/mainnet.yaml]: Returned back the value. Good catch 👍

**Comment by Nashatyrev** [comment]
Something weird happening with CI checks indication on github :/

**Comment by tbenr** [comment]
> Something weird happening with CI checks indication on github :/ 

If you are referring to the md check, its a broken link that has been fixed on master

**Pull request merged**